src/main/java/io/github/enkarin/bookcrossing/books/service/AttachmentService.java

```diff
@@ -0,0 +1,71 @@
+package io.github.enkarin.bookcrossing.books.service;
+
+import io.github.enkarin.bookcrossing.books.dto.AttachmentDto;
+import io.github.enkarin.bookcrossing.books.model.Attachment;
+import io.github.enkarin.bookcrossing.books.model.Book;
+import io.github.enkarin.bookcrossing.books.repository.AttachmentRepository;
+import io.github.enkarin.bookcrossing.books.repository.BookRepository;
+import io.github.enkarin.bookcrossing.errors.ErrorListResponse;
+import io.github.enkarin.bookcrossing.user.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.util.Locale;
+import java.util.Optional;
+
+@RequiredArgsConstructor
+@Service
+public class AttachmentService {
+
+    private final UserRepository userRepository;
+    private final AttachmentRepository attachRepository;
+    private final BookRepository bookRepository;
+
+    public ErrorListResponse saveAttachment(final AttachmentDto attachmentDto, final String login) throws IOException {
+        final ErrorListResponse response = new ErrorListResponse();
+        final Optional<Book> book = userRepository.findByLogin(login).orElseThrow().getBooks().stream()
+                .filter(b -> b.getBookId() == attachmentDto.getBookId())
+                .findFirst();
+        if(book.isPresent()){
```

Comment:
It looks that checkstyle doesn't work

Reply:
Checkstyle will be configured later, only PMD is enabled